### PR TITLE
[Issue:5687] Add prefix for new keys from Env

### LINKED ISSUE #5687: Failed to enable State API in K8s installation
**Describe the bug**
Can't enable State API in a minikube-based installation of Pulsar, despite adding the appropriate property `extraServerComponents` to the values-mini.yaml in the bookkeeper.configData section.

**To Reproduce**
1. Add the following line to values-mini.yaml under the bookkeeper.configData section:
```yaml
extraServerComponents: "org.apache.bookkeeper.stream.server.StreamStorageLifecycleComponent"
```
2. Launch pulsar cluster via helm (this is using the helm templates found in this repo):
```
helm install . --values values-mini.yaml helm -n pulsar
```
3. When the K8s cluster is ready, check the configuration of the bookkeeper nodes and confirm that the `extraServerComponents` property in `conf/bookkeeper.conf` is not defined with the specified value.

**Expected behavior**
That following the steps to reproduce the bookkeeper nodes would have this property defined.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Minikube (v1.4.0) deployment
 - Pulsar version v2.4.1

**Additional context**
I believe this issue is due to https://github.com/apache/pulsar/commits/master/docker/pulsar/scripts/apply-config-from-env.py which ignores commented lines, as such to fix this issue, the extraServerComponents property should not be commented but defined with a default empty value. 

This issue is not limited to the aforementioned property only but to all properties that are commented out, as the setup/procedure to enable such features are not working for K8s deployments.

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolong.ran <rxl@apache.org>

Fixes #5687 

### Motivation

When the user wants to add new keys for Env, adding fails if no prefix is added. 

Currently, add new keys for Env use the script of [apply-config-from-env.py](https://github.com/apache/pulsar/commits/master/docker/pulsar/scripts/apply-config-from-env.py), to ensure that the env set by the user can take effect, add the prefix(**PULSAR_PREFIX_**) for all keys.

### Modifications

- Add prefix for new keys from Env

### CODE REVIEW

File: deployment/kubernetes/generic/k8s-1-9-and-above/broker.yaml [hunk 1]
```diff
@@ -25,17 +25,17 @@ metadata:
...
     # change the managed ledger settings if you have more bookies
-    managedLedgerDefaultEnsembleSize: "1"
-    managedLedgerDefaultWriteQuorum: "1"
-    managedLedgerDefaultAckQuorum: "1"
+    PULSAR_PREFIX_managedLedgerDefaultEnsembleSize: "1"
+    PULSAR_PREFIX_managedLedgerDefaultWriteQuorum: "1"
+    PULSAR_PREFIX_managedLedgerDefaultAckQuorum: "1"
     # enable pulsar functions
-    functionsWorkerEnabled: "true"
-    PF_pulsarFunctionsCluster: local
```
Comment (jiazhai) [inline deployment/kubernetes/generic/k8s-1-9-and-above/broker.yaml]: There was already a prefix for pulsar functions: 
```PF_ENV_PREFIX = 'PF_'```
and `gen-yml-from-env.py` will handle it. we may not need to handle it.

Comment (wolfstudy) [reply deployment/kubernetes/generic/k8s-1-9-and-above/broker.yaml]: Yes, will fix it

File: deployment/kubernetes/aws/broker.yaml [hunk 2]
```diff
@@ -25,15 +25,15 @@ metadata:
...
-    PULSAR_GC: "\" -XX:+UseG1GC -XX:MaxGCPauseMillis=10\""
-    zookeeperServers: zk-0.zookeeper,zk-1.zookeeper,zk-2.zookeeper
-    configurationStoreServers: zk-0.zookeeper,zk-1.zookeeper,zk-2.zookeeper
-    clusterName: us-east
-    managedLedgerDefaultEnsembleSize: "2"
-    managedLedgerDefaultWriteQuorum: "2"
-    managedLedgerDefaultAckQuorum: "2"
-    deduplicationEnabled: "false"
+    PULSAR_PREFIX_PULSAR_MEM: "\" -Dio.netty.leakDetectionLevel=disabled -Dio.netty.recycler.linkCapacity=1024 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+AggressiveOpts -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC -XX:-ResizePLAB -XX:+ExitOnOutOfMemoryError -XX:+PerfDisableSharedMem -Xms12g -Xmx12g -XX:MaxDirectMemorySize=14g -Dpulsar.root.logger=DEBUG,FILE \""
```
Comment (cckellogg) [inline deployment/kubernetes/aws/broker.yaml]: There is no need to put a prefix in front of PULSAR_MEM, PULSAR_GC, or PULSAR_EXTRA_OPTS. This script (https://github.com/apache/pulsar/blob/master/conf/pulsar_env.sh) will pick up the environment variables. Same for all the other places.

Also, adding the PULSAR_PREFIX everywhere will add random variables to the configs. For example the broker is started with the following command
```
bin/apply-config-from-env.py conf/broker.conf &&
bin/apply-config-from-env.py conf/pulsar_env.sh &&
bin/gen-yml-from-env.py conf/functions_worker.yml &&
bin/pulsar broker
```
Every PULSAR_PREFX_ key value will end up in the conf/pulsar_env.sh script. That might not matter now or break things but I don't think is the correct approach in general for the broker or proxy configs. In my opinion the override scripts should be completely removed from broker and proxy and function workers. There are ways to apply configs without doing search and replaces with scripts and i think this is cleaner and less error prone.

For example, you can create a broker configmap with the configs you want to override:
data:
  broker.conf: |-
    zookeeperServers=zk-0,zk-1,zk-2
    configurationStoreServers=zk-0,zk-1,zk-2
    clusterName=my-cluster

This config can be mounted as a file in the container and then you can set this broker conf environment variable to where the file was mounted. For example:
PULSAR_BROKER_CONF: /conf/broker.conf 

Then you can just start the broker 
```
./bin/pulsar broker
```
The broker will read the config you mounted through the configmap and merge it with the default values.

If you want to add functions just add another config map with the function config and add the following flag to where the file was mounted.
```
./bin/pulsar broker --functions-worker-conf /conf/functions_worker.yml
```
The same can be done for the proxy. I think this makes the deployments easier and removes the search and replace scripts from the brokers and proxy. The approach also allows you add any config in the future without having to update a script.

Comment (wolfstudy) [reply deployment/kubernetes/aws/broker.yaml]: > There is no need to put a prefix in front of PULSAR_MEM, PULSAR_GC, or PULSAR_EXTRA_OPTS. This script (https://github.com/apache/pulsar/blob/master/conf/pulsar_env.sh) will pick up the environment variables. Same for all the other places.

ok, I will fix them.

> Every PULSAR_PREFX_ key value will end up in the conf/pulsar_env.sh script. That might not matter now or break things but I don't think is the correct approach in general for the broker or proxy configs.

Yes, this change is only used to fix problems encountered by users during the current use.

>In my opinion the override scripts should be completely removed from broker and proxy and function workers. There are ways to apply configs without doing search and replaces with scripts and i think this is cleaner and less error prone.

The proposal looks good to me,  this is a big change, maybe we can open a PIP to discuss further details.

**Comment by wolfstudy** [comment]
@cckellogg fixed done, remove `PULSAR_PREFIX_` from `PULSAR_MEM`, `PULSAR_GC` and `PULSAR_EXTRA_OPTS `.

**Comment by jiazhai** [comment]
run cpp tests
run java8 tests

**Comment by wolfstudy** [comment]
run cpp tests

File: deployment/kubernetes/aws/broker.yaml (hunk 2 above)
Comment (cckellogg) [reply deployment/kubernetes/aws/broker.yaml]: This change only helps if the user builds their own image or waits for a release image. This change still dumps unexpected values into config files that do no expect it. Does that matter? Changing the chart might be a slightly bigger change but it fixes the problem for users on past releases images.

**Pull request merged**